packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -101,16 +102,32 @@ class _DatePickerHeader extends StatelessWidget {
     switch (orientation) {
       case Orientation.portrait:
         height = _kDatePickerHeaderPortraitHeight;
-        padding = const EdgeInsets.symmetric(horizontal: 24.0);
+        padding = const EdgeInsets.symmetric(horizontal: 16.0);
         mainAxisAlignment = MainAxisAlignment.center;
         break;
       case Orientation.landscape:
         width = _kDatePickerHeaderLandscapeWidth;
-        padding = const EdgeInsets.all(16.0);
+        padding = const EdgeInsets.all(8.0);
         mainAxisAlignment = MainAxisAlignment.start;
         break;
     }
 
+    Widget yearButton = new _DateHeaderButton(
+      color: backgroundColor,
+      onTap: Feedback.wrapForTap(() => _handleChangeMode(_DatePickerMode.year), context),
+      child: new Text(new DateFormat('yyyy').format(selectedDate), style: yearStyle),
+    );
+    Widget dayButton = new _DateHeaderButton(
+      color: backgroundColor,
+      onTap: Feedback.wrapForTap(() => _handleChangeMode(_DatePickerMode.day), context),
+      child: new Text(new DateFormat('E, MMM\u00a0d').format(selectedDate), style: dayStyle),
+    );
+
+    if (mode == _DatePickerMode.day)
```

Comment:
This would be a good place for a comment that says something like: Only let a tap reach a button and cause an ink splash if it will change the mode.